src/zkevm_specs/evm/instruction.py

```diff
@@ -368,6 +377,42 @@ def range_check(self, value: Expression, n_bytes: int) -> bytes:
         except OverflowError:
             raise ConstraintUnsatFailure(f"Value {value} has too many bytes to fit {n_bytes} bytes")
 
+    def constrain_abs_word(self, x: RLC, x_abs: RLC, x_is_neg: FQ):
+        expected_abs, expected_is_neg = self.abs_word(x)
+        self.constrain_equal(self.word_is_equal(x_abs, expected_abs), FQ(1))
+        self.constrain_equal(x_is_neg, expected_is_neg)
+
+    # Return a tuple of `abs(x)` and `x_is_neg`.
+    def abs_word(self, x: RLC) -> Tuple[RLC, FQ]:
+        is_neg = self.word_is_neg(x)
+        is_zero = self.word_is_zero(x)
+        x_abs = self.select(
+            is_neg, self.rlc_encode((1 - is_zero.n) * (1 << 256) - x.int_value, 32), x
+        )
```

Comment:
Because this step is to generate the witness `x_abs`, we should directly use python `if` to check the condition.

```suggestion
        x_abs = x if is_neg == 0 else self.rlc_encode((1 - is_zero.n) * (1 << 256) - x.int_value, 32)
```